### PR TITLE
[Discover] Misc fixes

### LINKED ISSUE #5137: [Discover] Documents view is too small in Dashboards 2.10
**Is your feature request related to a problem? Please describe.**

Documents view has smaller effective area in 2.10 compared to previous versions

**Describe the solution you'd like**

Add option to hide the time histogram graph
OR
Make time histogram graph scroll with the documents like it used to

**Describe alternatives you've considered**

none

**Additional context**

In dashboards 2.10 time histogram graph is static on the page meaning the documents view (outlined in green) is effectively smaller than previous versions where the graph scrolled along with document list
dashboards2 10

Hiding the graph with custom css makes the document view (outlined in green) larger, allowing more documents to fit on a screen when the histogram is not relevant
dashboards2 10a

### PR DESCRIPTION
**Pull request by ashwin-pc**
### Description

Misc fixes:
1. Removes deprecation of legacy discover notice
2. Allows content below to still be interactive when the flyout is open
3. Allows canvas to scroll freely for discover.
4. Hides field action button unless hovered or focussed on

### Issues Resolved

fixes #5137 

## Screenshot

https://github.com/opensearch-project/OpenSearch-Dashboards/assets/20453492/cf54970f-4c0e-4ef1-972e-0f1ba4e2a9f1

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
  - [ ] `yarn test:ftr`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Comment by abbyhu2000** [comment]
not sure if some of the changes affect functional test result, could you rebase and see if ciGroup 2 and 3 pass? @ashwin-pc

**Comment by ashwin-pc** [comment]
Added some followup fixes from #5085

**Review by kavilla — APPROVED** [review]
Works for me!

File: src/plugins/discover/public/application/view_components/canvas/index.tsx [hunk 1]
```diff
@@ -43,39 +41,6 @@ export default function DiscoverCanvas({ setHeaderActionMenu, history }: ViewPro
     bucketInterval: {},
   });
 
-  const [isCallOutVisible, setIsCallOutVisible] = useState(
-    localStorage.getItem(KEY_SHOW_NOTICE) !== 'false'
```
Comment (kavilla) [inline src/plugins/discover/public/application/view_components/canvas/index.tsx:47]: it's probably unnecessary to clear out this key if it does exist correct?

Comment (ashwin-pc) [reply src/plugins/discover/public/application/view_components/canvas/index.tsx:47]: Yeah, thats overkill :)

**Pull request merged**